lib/rules.js

```diff
@@ -184,6 +227,185 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
 // e.g. |-3| -> 3
 export const ABSOLUTE_VALUE = defineRuleString('|-#a|', '#a')
 
+// ROOT
+
+// assumes that the exponent is a constant value
+// TODO: handle cases like nthRoot('x^2y', 2)
+// e.g. nthRoot(x^2, 4) -> nthRoot(x^1 ,2)
+export const CANCEL_EXPONENT = defineRule(
+    (node) => {
+        let isNthRoot = false
+        let validRadicand = false
+        if (query.isApply(node)) {
+            isNthRoot = (node.op == 'nthRoot')
+            validRadicand = query.isPow(node.args[0])
+        }
+        return (isNthRoot && validRadicand) ? {node} : null
```

Comment:
No there is no assumption. It works for any signed combination of exponent and index.